### ISSUE TITLE
Add more line-editing commands

### ISSUE THREAD

**Issue by cpavlina** [cpavlina/esh#2]
See the standard bash line-editing commands: http://web.mit.edu/gnu/doc/html/features_7.html

We could add more of these. They should be build options, as they're too big to justify inclusion on small microcontrollers. In particular, many support a sort of clipboard and will require the creation of an additional buffer.

**Comment by KeithoHimself** [comment]
This is really needed for embedded command interpreters, menus, etc...
Is this project still active?  I am looking at putting esh into a project.

**Comment by alexisvl** [comment]
Not really, I use it regularly but am not terribly interested in continuing to improve it as it meets my needs just fine. I'd happily review and merge a PR, though.
